client/client.go

```diff
@@ -48,23 +48,53 @@ var (
 // based implementation MUST return this error when it receives a 404 StatusCode.
 type Fetcher func(ctx context.Context, path string) ([]byte, error)
 
+// CheckpointFetcher is the signature of a function which can retrieve the latest
+// checkpoint from a log's data storage.
+type CheckpointFetcherFunc func(ctx context.Context) ([]byte, error)
+
+func CheckpointFetcher(f Fetcher) CheckpointFetcherFunc {
```

Comment:
Maybe it's a bit spicy, but I'd prefer the 3 functions to be something like:

```go
func FetcherToGetFuncs(f Fetcher) (GetCheckpointFunc, GetTileFunc, GetEntryBundleFunc) {
  return ....
}
```

At the point of use this would be one line instead of 3.